### PR TITLE
change how skip_if_exists is used

### PR DESCRIPTION
**Pull request by wjwwood**
See: https://github.com/ament/ament_tools/pull/56#discussion_r43160183

### CODE REVIEW

File: ament_tools/build_types/cmake.py [hunk 1]
```diff
@@ -216,6 +217,39 @@ def _common_cmake_on_test(self, context, build_type):
...
+
+        # Remove package level setup files so they can be replaced correctly either in the
+        # cmake install step or later with deploy_file(..., skip_if_exists=True)
+        for destination in destinations:
+            destination_path = compute_deploy_destination(
+                context,
+                os.path.basename(destination),
+                os.path.dirname(os.path.relpath(destination, context.build_space))
+            )
+            if os.path.exists(destination_path):
```
Comment (dirk-thomas) [inline ament_tools/build_types/cmake.py]: This is not enough if the `destination_path` is a link pointing to a non existing file.

File: ament_tools/helper.py [hunk 2]
```diff
@@ -222,6 +222,10 @@ def extract_argument_group(args, delimiting_option):
     return trimmed_args, extracted_args
 
 
+def compute_deploy_destination(context, filename, dst_subfolder=''):
+    return os.path.join(context.install_space, dst_subfolder, filename)
```
Comment (dirk-thomas) [inline ament_tools/helper.py]: This custom function only takes three arguments and passes them to `os.path.join` (except using `install_space`). Does this really have to be a separate function? I don't see a value over using `os.path.join` directly.

Comment (wjwwood) [inline ament_tools/helper.py]: It's used from two places. I did this incase the method of building this changed, then you'd be forced to change it in both locations.

Comment (dirk-thomas) [inline ament_tools/helper.py]: I don't see how it should change. It is constructing a path.

File: ament_tools/build_types/cmake.py (hunk 1 above)
Comment (wjwwood) [inline ament_tools/build_types/cmake.py]: Ok, now checks for `islink` too, and I changed it to use `os.remove`.

File: ament_tools/helper.py (hunk 2 above)
Comment (wjwwood) [inline ament_tools/helper.py]: Well how the path is constructed could change. I saw non-trivial construction of a value, being duplicated into two places and feared that someday someone might change one but not the other, so I made it into a function.

**Comment by wjwwood** [comment]
Are there any other outstanding issues?

**Comment by dirk-thomas** [comment]
Beside my previous comment : lgtm

**Pull request merged**